src/fns.c

```diff
@@ -3591,9 +3471,6 @@ compilation.  */);
   staticpro (&string_char_byte_cache_string);
   string_char_byte_cache_string = Qnil;
 
-  require_nesting_list = Qnil;
-  staticpro (&require_nesting_list);
```

Comment:
You don't seem to be protecting the variable from GC on the Rust side.

I've found some docs here: https://www.gnu.org/software/emacs/manual/html_node/elisp/Writing-Emacs-Primitives.html

Reply:
Aye. This probably explains the failures in CI that I do not see locally.

I am preparing a PR with a rewrite of the build.rs that will also handle protected statics. The current build code shows my lack of experience coding in Rust. Once that PR is blessed I will update this PR.

Reply:
I did see some interesting bugs when I prototyped porting `cons` (which I still need to fix). If you deliberately create a some large lists in a loop you can trigger a few GCs to flush out issues.

Reply:
Good suggestion. A similar large list is what is holding up the map PR.